### DISCUSSION TITLE
Parseq recognition model is repeating characters

### DISCUSSION THREAD

**Discussion by hanshupe007** [mindee/doctr#1745] [General]
After fine-tuning the parseq model is recognizing too many characters, in particular if the same character occurs multiple times in a row.
For example: O11001 is recognized as O111001,  -942.333 as -942.3333, : 66719 as 666719  or UOOO as UOOOO (the words are clearly readable in standard Arial font). Another example:

![image](https://github.com/user-attachments/assets/25f0c9a4-53d9-4a73-be37-b5659aef7b26)
73973 as 73973973

I trained with freeze-backbone on 25k synthetic data, and also tested (for now) on synthetic data. Did anyone experience similar issues?

**Comment by milosacimovic** [comment]
Yes, I found most models to be unreliable in terms of adding more characters after finetuning, especially if vocab is changed from the pretrained one.

**Comment by felixdittrich92** [comment]
Mh .. a while ago I fine tuned also `parseq` on a 100% synth dataset which includes additional chars for (dutch, spanish, norwegian, italian, german, polish, czech, finnish, danish, portuguese) without issues.
https://huggingface.co/Felix92/doctr-torch-parseq-multilingual-v1

One thing: You shouldn't use the doctr internal word generator - this is gold for testing/model debugging sessions but nothing for real fine tuning

I modified SynthTiger to my own needs https://github.com/felixdittrich92/synthtiger/tree/doctr-modified (branch: doctr-modified)
dataset: https://drive.google.com/file/d/1TNQN8uBMiGjzf2GM41BWDICefLubau5q/view?usp=drive_link

**Reply by felixdittrich92** [reply]
I think the best you can do would be to generate synth train data and a real world validation dataset (which can be small but should representative for what you need to reach)
